Sources/Recycling/ListViews/AdManagement/Cells/UserAdManagementButtonAndInformationCell.swift

```diff
@@ -0,0 +1,123 @@
+            button.accessibilityLabel = buttonText
+            updateButtonConstraints()
+        }
+    }
+    public var informationText: String? {
+        didSet {
+            informationLabel.attributedText = NSAttributedString(string: informationText ?? "")
+            updateInformationLabelConstraints()
+        }
+    }
+
+    private let buttonHeight: CGFloat = 32 // This is too small for comfort (Re. Apple's HIG), won't handle this now as the whole design is still subject to change
+    private let labelToButtonSpacing: CGFloat = .mediumSpacing
+    private let labelLeftInset: CGFloat = .mediumLargeSpacing
+    private let labelWidthProportion: CGFloat = 0.67
+
+    private lazy var informationLabel: UILabel = {
+        let label = UILabel(frame: .zero)
+        label.translatesAutoresizingMaskIntoConstraints = false
+        label.numberOfLines = 0
+        label.font = .title5
+        label.textColor = .licorice
+        label.textAlignment = .left
+        return label
+    }()
+
+    private lazy var button: UIButton = {
+        let button = UIButton(frame: .zero)
+        button.titleLabel?.font = .title5
+        button.titleLabel?.textColor = .milk
+        button.translatesAutoresizingMaskIntoConstraints = false
+        button.backgroundColor = .primaryBlue
+        button.layer.cornerRadius = 8
+        button.setContentCompressionResistancePriority(.defaultHigh, for: .horizontal)
+        button.setContentHuggingPriority(.defaultHigh, for: .horizontal)
+        return button
+    }()
+
+    // MARK: - init
+
+    override init(style: UITableViewCell.CellStyle, reuseIdentifier: String?) {
+        super.init(style: style, reuseIdentifier: reuseIdentifier)
+        selectionStyle = .none
+        button.addTarget(self, action: #selector(handleButtonTouchUpInside(_:)), for: .touchUpInside)
+        button.addTarget(self, action: #selector(handleButtonTouchDown(_:)), for: .touchDown)
+        setup()
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    // MARK: - ButtonActions
+
+    @objc private func handleButtonTouchUpInside(_ sender: UIButton) {
+        delegate?.buttonAndInformationCellButtonWasTapped(self)
+
+        UIView.animate(withDuration: 0.1, delay: 0, options: .beginFromCurrentState, animations: {
+            sender.backgroundColor = .primaryBlue
+        }, completion: nil)
+    }
+
+    @objc private func handleButtonTouchDown(_ sender: UIButton) {
+        sender.backgroundColor = .callToActionButtonHighlightedBodyColor
+    }
+
+    // MARK: - Constraint updates
+
+    private func updateButtonConstraints() {
+        guard let buttonText = buttonText else { return }
+
+        let buttonWidth = 20 + buttonText.width(withConstrainedHeight: buttonHeight, font: .title5)
+        if let widthConstraint = (button.constraints.filter { $0.firstAttribute == .width }).first {
```

Comment:
Can we have a property for the button constraint instead and then just modify the constant instead?